gui/widgets.py

```diff
@@ -296,3 +309,126 @@ def __update_data(self, data):
         self.line.set_data(self.time, self.data)
 
         return self.line,
+
+
+class SensorIndicator(tk.Frame):
+    """ TKinter frame that holds a TKinter square of color and a label for sensor.name
+
+    The box changes color depending on status of sensor. The sensor must have a self.status attribute.
+
+    Parameters
+    ----------
+    parent : TKinter Frame
+        parent frame
+    gateway : Gateway instance
+        Gateway to monitor
+    sensor : attribute of a Sensors instance
+        sensor to display status from
+    field : str
+        name of the sensor to display
+
+    """
+
+    def __init__(self, parent, gateway, sensor, field, *args, **kwargs):
+        tk.Frame.__init__(self, parent, *args, **kwargs)
+        self.parent = parent
+        self.gateway = gateway
+        self.sensor = sensor
+        self.field = field
+
+        # Button to make a colored "box" for sensor
+        # Style will be reflected on this button
+        self.btn = tk.Button(self, text='', bg='red', state=tk.DISABLED, width=20)
+        self.btn = tk.Button(self)
+        self.btn.grid(row=0, column=1)
+        # Label to display the gateway's port name
+        self.label = tk.Label(self, text=self.field)
+        self.label.grid(
+            row=0, column=2, padx=5)
+
+        self.__update_button()
+
+    def __update_button(self):
+        """ Set the style of the button depending on the status of sensor.
+
+        """
+        if self.sensor.got_calibration:
+            #if self.field == "Telemetry Status":
+            #    self.btn.config(bg='green', height=2, width=3, state=tk.DISABLED)
+            #    self.sensor.got_calibration = False
+            #else:
+            self.btn.config(bg='green', height=1, width=2, state=tk.DISABLED)
+            self.sensor.got_calibration = False
+        else:
+            #if self.field == "Telemetry Status":
+            #    self.btn.config(bg='red', height=2, width=3, state=tk.DISABLED)
+            #    self.sensor.got_calibration = True
+            # else:
+            self.btn.config(bg='red', height=1, width=2, state=tk.DISABLED)
+            self.sensor.got_calibration = True
+        # Call this function again after 500 ms
+        # Is this needed for the button? How often will this be checked? Just after calibration?
```

Comment:
The event "calibration received" happens in the background (in the `Gateway` class). The GUI can't know when this happens so we must check regularly